addon/i3dio/utility.py

```diff
@@ -81,23 +82,23 @@ def as_export_path(filepath: str) -> str:
         filepath (str): The path to the file, as used or stored by/in Blender.
 
     Returns:
-        str: The resolved export path (either $data-relative, relative to blend file, or absolute).
+        Path: The resolved path, either as a relative path (to the blend file) or an absolute path.
     """
     if filepath.startswith('$data'):
-        # If the path already starts with $data, return it as is (can happen from certain shader textures)
-        return filepath
-    fs_path = as_fs_relative_path(filepath)
-    if fs_path.startswith('$data'):
+        # Already $data-prefixed (can happen from certain shader textures)
+        return Path(filepath)
+
+    # Check if inside FS data directory
+    if (fs_path := as_fs_relative_path(filepath)).parts and fs_path.parts[0] == '$data':
         return fs_path
-    # If under blend file directory, use blend-relative path
+
+    # Try to make path relative to the .blend file
     blend_dir = Path(bpy.data.filepath).parent.resolve()
     target_path = Path(bpy.path.abspath(filepath)).resolve(strict=False)
     try:
-        rel = target_path.relative_to(blend_dir)
-        return rel.as_posix()  # Relative to blend file directory
+        return Path(os.path.relpath(str(target_path), str(blend_dir)))

```

Comment:
```suggestion
        # PurePath.relative_to broke until 3.12 https://docs.python.org/3.12/library/pathlib.html#pathlib.PurePath.relative_to
        return Path(os.path.relpath(str(target_path), str(blend_dir)))
```
This will actually be "fixed" from Python 3.12 and onward by the addition of a "walk_up" parameter so we don't need to resort to using the OS module although it can still fail on Windows if the files are on different drives as the path would no longer be relative by then. Unfortunately the new LTS is still on python 3.11, so for now all we can do is leave a comment in case we forget and get confused about the use of `os.path.relpath`  🥲

Reply:
Ahhha... Blender follow VFX Reference Platform and the plan seems to be python 3.13 in 2026😅